src/common/commonutils/Internal.h

```diff
@@ -42,4 +44,15 @@
 
 #define MAX_STRING_LENGTH 512
 
+// Buffer sizes for string conversion of numeric values
+// Based on maximum possible digits for each type plus sign and null terminator
+#define MAX_INT_STRING_LENGTH 16    // Accommodates 32-bit int values
+#define MAX_LONG_STRING_LENGTH 32   // Accommodates 64-bit long values
+
+#define OSConfigTelemetryStatusTrace(handle, callingFunctionName, status) do { char status_str[MAX_INT_STRING_LENGTH] = {0}; snprintf(status_str, sizeof(status_str), "%d", (status)); char line_str[MAX_INT_STRING_LENGTH] = {0}; snprintf(line_str, sizeof(line_str), "%d", __LINE__); const char* keyValuePairs[] = { "Filename", __FILE__, "LineNumber", line_str, "FunctionName", __func__, "CallingFunctionName", (callingFunctionName) ? (callingFunctionName) : "-", "ResultCode", status_str, "DistroName", GetOsName(NULL, handle), "CorrelationId", getenv("activityId") ? getenv("activityId") : "", "Version", OSCONFIG_VERSION }; OSConfigTelemetryLogEvent((handle), "StatusTrace", keyValuePairs, sizeof(keyValuePairs) / sizeof(keyValuePairs[0]) / 2); } while(0)
```

Comment:
It looks like we are missing `RuleCodename` here from the specification and using a hard-coded string instead. It should be the object name (like "auditEnsureAuthenticationRequiredForSingleUserMode") so that we can derive insights (such as "who called `ExecuteCommand`?").

I understand that's bulky to bring down to the common utilities but that is what we signed off on. Let me know if I misunderstood what we are doing here.